### DISCUSSION TITLE
Clicking on a dropdown menu item only registers after second click (on first load of the page)

### DISCUSSION THREAD

**Discussion by botacode-gh** [neuefische/questions#605] [Web] [answered]
When the page first loads, entering something into the text input field and then clicking on a result in the dropdown menu does nothing. Only after repeating the process a second time, the click gets registered (and the form gets submitted as expected). 
This happens only once, when the page first loads. After that, it works as expected.

This bug can be replicated [on Vercel](https://capstone-project-qj71zd3g9-botacode-gh.vercel.app/) following the steps above. 
Here's [the code](https://github.com/botacode-gh/capstone-project/pull/19/files#diff-87207e3a729e2be3d5ae1c96676c9dcbcfea1b8fde0d0eec90789a64f99ce990).

A picture is worth a thousand words, and my video capture's gotta have at least 10 pictures:
https://github.com/neuefische/questions/assets/124698651/f8df0ab2-ebc1-4fdf-a46a-5c10077217a6

**Comment by moonwave99** [comment, marked as answer]
I would split the submit and click behaviours:

```js
// HomePage.js
function handleInput(name) {
    const results = searchItems(name);
    const firstResult = results[0];
    if (!results.length) {
      return;
    }

    if (localStorage.getItem("acquiredItems")) {
      const acquiredItems = JSON.parse(localStorage.getItem("acquiredItems"));

      const acquiredItemNames = acquiredItems.map((item) => item.name);

      if (acquiredItemNames.includes(firstResult.name)) {
        setItemAlreadyAcquired(true);
        return;
      }
    }
    if (results.length > 0) {
      addAcquiredItem(firstResult);
      setAddedItem(firstResult);
      setInputPrompt(`Added ${firstResult.name} to your island!`);
      setModalMessage(`Added ${firstResult.name} to your island!`);
      setModalIsVisible(!modalIsVisible);
    }
}

const handleSubmit = (name) => {
    handleInput(name)
};

const handleClick = (result) => {
    handleInput(result.name);
};

...

<NewItemForm
    handleSubmit={handleSubmit}
    handleClick={handleClick}
    allItems={allItems}
    modalIsVisible={modalIsVisible}
/>
```

Then I would move the `inputRef` inside the `NewItemForm`, and there:

- call `handleClick(result)` when the user clicks a suggestion;
- call `handleSubmit(inputRef.current.value)` when the user submits the form.

Remember to define the `handleClick` prop inside `NewItemForm` as well!
